samples/secure-mcp-sse-client-server/README.md

```diff
@@ -0,0 +1,89 @@
+
+By default, Quarkus GitHub provider submits the client id and secret in the HTTP Authorization header.
+However, GitHub may require that both client id and secret are submitted as form parameters instead.
+
+When you get HTTP 401 error after logging in to GitHub and being redirected back to Quarkus MCP server,
+try to replace `quarkus.oidc.credentials.secret=${github.client.secret}` property
+with the following two properties instead:
+
+```properties
+%quarkus.oidc.credentials.client-secret.method=post
+%quarkus.oidc.credentials.client-secret.value=${github.client.secret}
+```
+
+# AI Gemini API key
+
+Get [AI Gemini API key](https://aistudio.google.com/app/apikey) and set the `quarkus.langchain4j.ai.gemini.api-key` property.
+
+# Running the sample in dev mode
+
+### MCP server
+
+Start the mcp server component in the `secure-mcp-server` directory using `mvn quarkus:dev -Duser.name="your-github-account-name"`, for example: `mvn quarkus:dev -Duser.name="John Doe"`.
+
+Note, you should use double quotes to register your GitHub account's full name
+
+This will start the server on port 8081 and create a database containing a record with your GitHub account name and a `read:name` permission associated with this name.
+
+### MCP Client
+
+Make sure the GitHub OAuth2 application is registered, AI Gemini API key is available and the MCP server is started.
+
+Start the client component in the `secure-mcp-client` directory using `mvn quarkus:dev`.
+
+Go to `http://localhost:8080`, login with GitHub, request a poem and confirm that your name is returned alongside the poem about Java.
+
+# Running the sample in prod mode
+
+### MCP server
+
+Start Postgresql:
+
+```shell
+docker run --rm=true --name quarkus_test -e POSTGRES_USER=quarkus_test -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test -p 5432:5432 postgres:17.0
+```
+
+Package the mcp server component in the `secure-mcp-server` directory using `mvn clean package -Duser.name="your-github-account-name"`, for example: `mvn clean package -Duser.name="John Doe"`.
+
+Note, you should use double quotes to register your GitHub account's full name
+
+And run it:
+
+```shell
+java -jar target/quarkus-app/quarkus-run.jar
+```
+
+This will start the server on port 8081 and create a database containing a record with your GitHub account name and a `read:name` permission associated with this name.
+
+### MCP Client
+
+Make sure the GitHub OAuth2 application is registered, AI Gemini API key is available and the MCP server is started.
+
+Package and run the mcp client component:
+
+```shell
+mvn clean package
+java -jar target/quarkus-app/quarkus-run.jar
+```
+
+Go to `http://localhost:8080`, login with GitHub, request a poem and confirm that your name is returned alongside the poem about Java.
+
+# Securiy
+
+Both Quarkus MCP server main SSE and tool endpoints require authentication.
```

Comment:
"tool endpoint" may be a bit misleading term (I usually say "POST" endpoint), and will be rendered obsolete when we move to the streamable HTTP transport. Could we rather say "All HTTP requests to the MCP server require authentication"? That sounds safer to me.

Then below, you could say that "the endpoint that serves for tool calling" requires the `read:name` permission.